demos/python_demos/bert_question_answering_embedding_demo/bert_question_answering_embedding_demo.py

```diff
@@ -0,0 +1,338 @@
+
+sys.path.append(os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'common'))
+from tokens_bert import text_to_tokens, load_vocab_file
+from html_reader import get_paragraphs
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    args = parser.add_argument_group('Options')
+    args.add_argument('-h', '--help', action='help', default=SUPPRESS, help='Show this help message and exit.')
+    args.add_argument("-i", "--input", help="Required. Urls to a wiki pages with context",
+                      required=True, type=str)
+    args.add_argument("--par_num", help="Optional. Number of contexts filtered in by embedding vectors",
+                      default=10, required=False, type=int)
+    args.add_argument("-v", "--vocab", help="Required. Path to vocabulary file with tokens",
+                      required=True, type=str)
+    args.add_argument("-m_emb","--model_emb", help="Required. Path to an .xml file with a trained model to build embeddings",
+                      required=True, type=str, default=None)
+
+    args.add_argument("-m_qa","--model_qa", help="Optional. Path to an .xml file with a trained model to give exact answer",
+                      required=False, type=str)
+    args.add_argument("--input_names", help="Optional. Names for inputs in MODEL_QA network. For example 'input_ids,attention_mask,token_type_ids','position_ids'",
+                      required=False, type=str)
+    args.add_argument("--output_names", help="Optional. Names for outputs in MODEL_QA network. For example 'output_s,output_e'",
+                      required=False, type=str)
+    args.add_argument("-a", "--max_answer_token_num", help="Optional. Maximum number of tokens in exact answer",
+                      default=15, required=False, type=int)
+    args.add_argument("-d", "--device",
+                      help="Optional. Specify the target device to infer on; CPU is "
+                           "acceptable. Sample will look for a suitable plugin for device specified. Default value is CPU",
+                      default="CPU", type=str)
+    args.add_argument('-c', '--colors', action='store_true',
+                      help="Optional. Nice coloring of the questions/answers. "
+                           "Might not work on some terminals (like Windows* cmd console)")
+    return parser
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    log.info("Creating Inference Engine")
+    ie = IECore()
+
+    #read model to calculate mebedding
+    model_xml_emb = args.model_emb
+    model_bin_emb = os.path.splitext(model_xml_emb)[0] + ".bin"
+
+    log.info("Loading embedding network files:\n\t{}\n\t{}".format(model_xml_emb, model_bin_emb))
+    ie_encoder_emb = ie.read_network(model=model_xml_emb, weights=model_bin_emb)
+    # check input and output names
```

Comment:
The actual check is missed. I also think keys similar to `--input_names`/`--output_names` should be added for embeddings net.

Reply:
the output_names do not make sense becasue embedding network has only one single output. There is no need to point output name.
input_names_emb will be added